src/client/pydaos/__init__.py

```diff
@@ -45,6 +45,7 @@ class DaosClient():
     the lifetime of a process.
     """
     _instance = None
+    noeq = 0
```

Comment:
I find it better to avoid negatives in code names as that avoids double negatives when comparing them which can be confusing.  As this isn't intended to be part of the full API then it should be prefixed with a `_` as well.
```suggestion
    _eq = True
```